crates/ruma-common/tests/api/ui/removed-without-deprecated.stderr

```diff
@@ -0,0 +1,14 @@
+error: no rules expected the token `history_impl`
+  --> tests/api/ui/removed-without-deprecated.rs:3:28
+   |
+3  |   const METADATA: Metadata = metadata! {
+   |  ____________________________^
+4  | |     method: GET,
+5  | |     rate_limited: false,
+6  | |     authentication: None,
+...  |
+10 | |     }
+11 | | };
+   | |_^ no rules expected this token in macro call
+   |
+   = note: this error originates in the macro `$crate::metadata` which comes from the expansion of the macro `metadata` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Comment:
I'm not entirely happy with this error, i feel like the `+` on the following line is making sure that it does not parse on the macro-side, but that way, it's losing out on a rich error from the `const fn`:
https://github.com/ruma/ruma/blob/9cb5934866e91091cd590458bb21e87f20c3108a/crates/ruma-common/src/api.rs#L477

Personally, i'd say to change this from `+` to `*`, to let it pass through, and hit the error on `const fn`, so that this is much richer and would explain what's happening better to the programmer.

Reply:
I just noticed your comment about `const fn` not being plenty useful with `trybuild`, I'd say to include this bit in your/the issue, so that we can make this error richer in the future.